internal/dataplane/configfetcher/kongrawstate.go

```diff
@@ -0,0 +1,318 @@
+					sanitizeAuth(jwtAuth)
+					kongState.Consumers[i].JWTAuths = append(kongState.Consumers[i].JWTAuths,
+						&kongstate.JWTAuth{
+							JWTAuth: *jwtAuth,
+						},
+					)
+				}
+			}
+		}
+		for _, basicAuth := range rawstate.BasicAuths {
+			if basicAuth.Consumer != nil {
+				if *basicAuth.Consumer.ID == *consumer.ID {
+					sanitizeAuth(basicAuth)
+					kongState.Consumers[i].BasicAuths = append(kongState.Consumers[i].BasicAuths,
+						&kongstate.BasicAuth{
+							BasicAuth: *basicAuth,
+						},
+					)
+				}
+			}
+		}
+		for _, aclGroup := range rawstate.ACLGroups {
+			if aclGroup.Consumer != nil {
+				if *aclGroup.Consumer.ID == *consumer.ID {
+					sanitizeAuth(aclGroup)
+					kongState.Consumers[i].ACLGroups = append(kongState.Consumers[i].ACLGroups,
+						&kongstate.ACLGroup{
+							ACLGroup: *aclGroup,
+						},
+					)
+				}
+			}
+		}
+		for _, oauth2Cred := range rawstate.Oauth2Creds {
+			if oauth2Cred.Consumer != nil {
+				if *oauth2Cred.Consumer.ID == *consumer.ID {
+					sanitizeAuth(oauth2Cred)
+					kongState.Consumers[i].Oauth2Creds = append(kongState.Consumers[i].Oauth2Creds,
+						&kongstate.Oauth2Credential{
+							Oauth2Credential: *oauth2Cred,
+						},
+					)
+				}
+			}
+		}
+		for _, mTLSAuth := range rawstate.MTLSAuths {
+			if mTLSAuth.Consumer != nil {
+				if *mTLSAuth.Consumer.ID == *consumer.ID {
+					sanitizeAuth(mTLSAuth)
+					kongState.Consumers[i].MTLSAuths = append(kongState.Consumers[i].MTLSAuths,
+						&kongstate.MTLSAuth{
+							MTLSAuth: *mTLSAuth,
+						},
+					)
+				}
+			}
+		}
+	}
+
+	return kongState
+}
+
+func rawPluginsToPlugins(plugins []*kong.Plugin) []kong.Plugin {
+	if len(plugins) == 0 {
+		return nil
+	}
+	ps := []kong.Plugin{}
+
+	for _, p := range plugins {
+		ps = append(ps, sanitizePlugin(*p))
+	}
+	return ps
+}
+
+func rawTargetsToTargets(targets []*kong.Target) []kongstate.Target {
+	if len(targets) == 0 {
+		return nil
+	}
+	ts := []kongstate.Target{}
+
+	for _, t := range targets {
+		ts = append(ts, kongstate.Target{Target: *t})
+	}
+	return ts
+}
+
+func rawCertificatesToCertificates(certificates []*kong.Certificate) []kongstate.Certificate {
+	if len(certificates) == 0 {
+		return nil
+	}
+	certs := []kongstate.Certificate{}
+
+	for _, c := range certificates {
+		certs = append(certs, kongstate.Certificate{
+			Certificate: sanitizeCertificate(*c),
+		})
+	}
+	return certs
+}
+
+func rawCACertificatesToCACertificates(caCertificates []*kong.CACertificate) []kong.CACertificate {
+	if len(caCertificates) == 0 {
+		return nil
+	}
+	certs := []kong.CACertificate{}
+
+	for _, c := range caCertificates {
+		certs = append(certs, sanitizeCACertificate(*c))
+	}
+	return certs
+}
+
+// -----------------------------------------------------------------------------
+// Sanitization functions
+// -----------------------------------------------------------------------------
+
+func sanitizeKongService(service kong.Service) kong.Service {
+	service.ID = nil
```

Comment:
I started wondering - why do we have to discard the entities' IDs? I believe that can break the stability of IDs when we use the last good known config. We fill the IDs in the [parser](https://github.com/Kong/kubernetes-ingress-controller/blob/ed0bdc60ddd02543e42e46ce75f92b43d029ffab/internal/dataplane/parser/parser.go#L252) and while we use the last good known config, this is not called.

Reply:
I wasn't aware of that functionality we have. By default, the IDs are not assigned by the controlplane, and because of that, I removed all of them. I think we should assign IDs when the FG is enabled for the last config as well. I'll do that

Reply:
2d5f0b76b544e50aff3380373ff2bce22b2632c6